stl/inc/chrono

```diff
@@ -3455,10 +3455,10 @@ namespace chrono {
         }
 
         enum class _LeapSecondRep : unsigned int {
-            _None, // tai_clock, gps_clock; oblivious to leap seconds
-            _Negative, // system_clock, observes negative, but not positive, leap seconds
+            _None, // system_clock, tai_clock, gps_clock; oblivious to leap seconds
+            _Negative, // TRANSITION, ABI: old value for system_clock

```

Comment:
After extensive thought and maximum caffeination, I believe that this can be improved - we shouldn't need to keep a TRANSITION, ABI value around forever.

The key is that although this `enum class` is passed between different functions, it never appears in anything that's separately compiled, and it doesn't appear as a data member anywhere. The only uses are within `_Make_time_point()` and all of its callers, like:

https://github.com/microsoft/STL/blob/17fde2cbab6e8724d81c9555237c9a623d7fb954/stl/inc/chrono#L4401

Because this is C++ mangled, it'll be sufficient to change the `enum class`'s name (which is an opportunity to follow the correct `_Ugly_snake` convention; otherwise we'd add a `2` suffix). This prevents any ABI problems - old code will be looking for the old `_LeapSecondRep` mangled names (and get the old instantiations that are aware of `_Negative`), while new code will be looking for the new `_Leap_second_rep` mangled names (and get the new instantiations that don't talk about `_Negative`).

In fact, renaming may even be necessary for ABI correctness - because the new code doesn't mention `_Negative` directly at all, if a severe mix-and-match scenario ends up passing `_Negative` to this code where the `if (_Leap == _LeapSecondRep::_Negative && _Tzdb._All_ls_positive)` etc. logic has been removed, I am unsure what will happen (I would need to step through the code in detail). By renaming, we don't have to worry about that scenario at all.

I'll validate and push changes.